{

'whitelist_statuses' => [
    'STATUS_ANY_ERROR',
],

'whitelist_patterns' => [
    # We have no special focus.
    [ 'MDEV-21669' , 'indexes inside InnoDB, which is different from the number of indexes' ],
],

'blacklist_statuses' => [
    'STATUS_OK',
],

'blacklist_patterns' => [
    # [ <search_pattern> , <description like MDEV-nnnn> ]
    # ---------------------------------------------------
    # RQG runs showing this pattern end frequent with the status
    # STATUS_ENVIRONMENT_FAILURE(110). In case its a grammar maintained by hand than the
    # grammar is faulty and should be corrected.
    # But grammar simplification and also grammar masking run some automatic mangling of
    # grammars which could lead to the same effect because of non avoidable and harmless
    # reason. So we throw the remainings of such runs away and save by that resources.
    [ 'bad_grammar' , 'Possible endless loop in grammar.'                                         ],
    #
    # Grammars could have the unfortunate runtime property to cause some dramatic growth
    # of storage space consumption. This is especially valid in case of grammars generated
    # - during test simplification
    # - via grammar masking.
    # Example: All DROPs and DELETEs were shrinked away.
    #
    # The rqg_batch.pl ResourceControl will try to prevent negative consequences by stopping
    # selected RQG runs and freeing the resources. But in case the speed of the changes
    # exceeds the capabilities of ResourceControl or ResourceControl is disabled than
    # some typical outcome (observed before ResourceControl was implemented) is:
    # The DB server asserts, threads and/or periodic reporters exit with (hopefully)
    # STATUS_SERVER_CRASHED, the reporter Backtrace prints content of the server error log
    # and lets some debugger analyze the core file and than RQG exits with (again hopefully)
    # STATUS_SERVER_CRASHED.
    # Than in case of some "unsharp" setup of the whitelists (statuses == STATUS_ANY_ERROR,
    # patterns empty) that all gets valuated as replay.
    # But its either
    # - (most likely) a replay of some known "natural" effect which is not a defect
    # or
    # - (quite unlikely) the replay of a defect but the surviving data are not sufficient
    #   for analysis and we should better not waste time on that.
    # Conclusion:
    # Runs where we hit trouble with storage space are "unwanted".
    # Server error log pattern: ..... (Error 28 is OS: No space left on device)
    # [ERROR] InnoDB: preallocating 12582912 bytes for file ./test/#sql-1f36d_10.ibd failed with error 28
    # [ERROR] InnoDB: preallocating 65536 bytes for file ./test/FTS_0000000000000524_00000000000005f9_INDEX_5.ibd failed with error 28
    [ 'bad_environment_1' , '\[ERROR\] InnoDB: preallocating .{1,120} with error 28'              ],
    # Debugger output pattern:
    #    BFD: Warning: <core file> is truncated: expected core file size >= 959647744, found: 55398400.
    [ 'bad_environment_2' , '  is truncated: expected core file size '                            ],
    #
    #### Open bugs (TBR == to be reported or already in JIRA, MDEV == in JIRA)
    [ 'TBR-50'     , 'mysqld: .{1,150}rem0rec.cc:.{1,30} ulint rec_get_converted_size_comp_prefix_low.{1,300}: Assertion .field->col->is_dropped.. .. .field->col->mbmaxlen .. len >= field->col->mbminlen . fixed_len . field->col->mbmaxlen. failed.' ],
    [ 'TBR-51'     , 'mysqld: .{1,150}row0sel.cc:.{1,30} dberr_t row_search_mvcc.{1,150}: Assertion .0. failed.' ],
    #  mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/row/row0sel.cc:5163: dberr_t row_search_mvcc(byte*, page_cur_mode_t, row_prebuilt_t*, ulint, ulint): Assertion `index != clust_index || row_get_rec_trx_id(rec, index, offsets)' failed.
    [ 'TBR-53'     , 'mysqld: .{1,150}row0sel.cc:.{1,30} dberr_t row_search_mvcc.{1,150}: Assertion .index \!= clust_index \|\| row_get_rec_trx_id\(rec, index, offsets\). failed.' ],
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/btr/btr0cur.cc:4139: void btr_cur_upd_rec_in_place(rec_t*, const dict_index_t*, const offset_t*, const upd_t*, buf_block_t*, mtr_t*): Assertion `!rec_offs_nth_sql_null(offsets, n)' failed.
    [ 'TBR-55'     , 'mysqld: .{1,150}btr0cur.cc:.{1,30} void btr_cur_upd_rec_in_place.{1,150}: Assertion .\!rec_offs_nth_sql_null\(offsets, n\). failed.' ],
    # Need Test
    [ 'TBR-56'     , 'mysqld:.{1,150}handler0alter.cc:.{1,30}: bool innobase_rename_or_enlarge_column_try.{1,150}: Assertion .\!\(\(col->prtype \^ prtype\) \& \~\(16384U\|32768U\)\). failed' ],
    #
    [ 'MDEV-17843' , 'mysqld: .{1,150}lock0lock.cc:.{1,30} bool lock_rec_queue_validate\(.{1,250}\): Assertion .page_rec_is_leaf\(rec\). failed.' ],
    # Fix is on move up
    [ 'MDEV-18911' , 'mysqld: .{1,150}row0sel.cc:.{1,30} void row_sel_field_store_in_mysql_format_func.{1,250}: Assertion .\(templ->is_virtual \&\& \!field\) \|\| \(field \&\& field->prefix_len \? field->prefix_len == len : templ->mysql_col_len == len\). failed.' ],
    [ 'MDEV-19130' , 'mysqld:.{1,150}handler.cc:.{1,30}: int handler::update_auto_increment\(\): Assertion .next_insert_id >= auto_inc_interval_for_cur_row.minimum\(\). failed' ],
    # Needs replay test
    [ 'MDEV-21325' , 'mysqld: .{1,150}trx0rec.cc:.{1,30} byte\* trx_undo_log_v_idx.{1,300}: Assertion .n_idx > 0. failed.' ],
    # 2020-02-05T20:29:25 [282821] #3  <signal handler called>
    # 2020-02-05T20:29:25 [282821] #4  0x00005653c2e1c3d0 in init_fts_doc_id_for_ref (table=table@entry=0x7fb06c074150, depth=depth@entry=0x7fb10cf28018) at /work/10.4/storage/innobase/row/row0mysql.cc:1774
    # 2020-02-05T20:29:25 [282821] #5  0x00005653c2e1f267 in init_fts_doc_id_for_ref (depth=0x7fb10cf28018, table=0x7fb06c074150) at /work/10.4/storage/innobase/row/row0mysql.cc:1757
    # 2020-02-05T20:29:25 [282821] #6  row_update_for_mysql
    [ 'TBR-200'    , '<signal handler called>.{1,500}init_fts_doc_id_for_ref.{1,500}init_fts_doc_id_for_ref.{1,500}row_update_for_mysql' ],
    [ 'TBR-201'    , 'InnoDB: Failing assertion: index_cache->words == NULL' ],
    # 2020-02-05T20:35:17 [264665] #3  <signal handler called>
    # 2020-02-05T20:35:17 [264665] #4  dict_col_t::detach (index=..., this=0x0) at /work/10.4/storage/innobase/include/dict0mem.h:1333
    # 2020-02-05T20:35:17 [264665] #5  dict_index_t::detach_columns (this=0x7fd9000d7670) at /work/10.4/storage/innobase/include/dict0mem.h:1192
    # 2020-02-05T20:35:17 [264665] #6  dict_mem_index_free
    [ 'TBR-202'    , '<signal handler called>.{1,500}dict_col_t::detach.{1,500}dict_index_t::detach_columns.{1,500}dict_mem_index_free' ],
    [ 'TBR-203'    , 'InnoDB: Failing assertion: strchr\(table->name.m_name, \'\/\'\) \!= NULL' ],
    # Looks like MDEV-21137 (has MTR tests) but that uses FOREIGN KEYs
    # mysqld: /work/10.4/storage/innobase/buf/buf0buf.cc:4840: buf_block_t* buf_page_get_gen(page_id_t, ulint, ulint, buf_block_t*, ulint, const char*, unsigned int, mtr_t*, dberr_t*): Assertion `mode == 16 || mode == 12 || !fix_block->page.file_page_was_freed' failed.
    [ 'TBR-204'    , 'mysqld: .{1,150}buf0buf.cc:.{1,30} buf_block_t\* buf_page_get_gen.{1,500}: Assertion .mode == 16 \|\| mode == 12 \|\| \!fix_block->page.file_page_was_freed. failed.' ],
    # mysqld: /work/10.4/storage/innobase/btr/btr0pcur.cc:157: void btr_pcur_store_position(btr_pcur_t*, mtr_t*): Assertion `page_get_page_no(block->frame) == index->page' failed.
    [ 'TBR-205'    , 'mysqld: .{1,150}btr0pcur.cc:.{1,30} void btr_pcur_store_position.{1,500}: Assertion .page_get_page_no\(block->frame\) == index->page. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353
    # mysqld: /home/mleich/work/bb-10.5-MDEV-12353/storage/innobase/mtr/mtr0mtr.cc:619: ulint mtr_t::prepare_write(): Assertion `n_recs > 0' failed.
    [ 'TBR-206'    , 'mysqld: .{1,150}mtr0mtr.cc:.{1,30} ulint mtr_t::prepare_write\(\): Assertion .n_recs > 0. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353
    # mysqld: /home/mleich/work/bb-10.5-MDEV-12353/storage/innobase/include/dyn0buf.h:311: ulint mtr_buf_t::size() const: Assertion `total_size == m_size' failed.
    [ 'TBR-207'    , 'mysqld: .{1,150}dyn0buf.h:.{1,30} ulint mtr_buf_t::size\(\) const: Assertion .total_size == m_size. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353
    # mysqld: /home/mleich/work/bb-10.5-MDEV-12353/storage/innobase/include/dict0dict.ic:818: dict_field_t* dict_index_get_nth_field(const dict_index_t*, ulint): Assertion `index->magic_n == 76789786' failed.
    [ 'TBR-208'    , 'mysqld: .{1,150}dict0dict.ic:.{1,30} dict_field_t\* dict_index_get_nth_field.{1,500}: Assertion .index->magic_n == 76789786. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353    but outside of InnoDB
    # # 2020-02-07T13:18:09 [139904] #3  <signal handler called>
    # 2020-02-07T13:18:09 [139904] #4  __memcmp_sse4_1 () at ../sysdeps/x86_64/multiarch/memcmp-sse4.S:70
    # 2020-02-07T13:18:09 [139904] #5  0x00005570f1d06145 in native_compare (length=0x7fada03d8dd0, a=0x7fabc05ed888, b=0x7fabc05ed818) at /home/mleich/work/bb-10.5-MDEV-12353/mysys/ptr_cmp.c:49
    # 2020-02-07T13:18:09 [139904] #6  0x00005570f1cf1b63 in my_qsort2 (base_ptr=0x7fabc05ed818, count=15, size=8, cmp=0x5570f1d06111 <native_compare>, cmp_argument=0x7fada03d8dd0) at /home/mleich/work/bb-10.5-MDEV-12353/mysys/mf_qsort.c:154
    # 2020-02-07T13:18:09 [139904] #7  0x00005570f1620b5e in Filesort_buffer::sort_buffer
    [ 'TBR-209'    , '<signal handler called>.{1,500}__memcmp_sse4_1.{1,500}native_compare.{1,500}my_qsort2' ],
    # 2020-02-07 bb-10.5-MDEV-12353 after update
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/include/dict0dict.ic:817: dict_field_t* dict_index_get_nth_field(const dict_index_t*, ulint): Assertion `pos < index->n_def' failed.
    [ 'TBR-210'    , 'mysqld: .{1,150}dict0dict.ic:.{1,30} dict_field_t\* dict_index_get_nth_field.{1,500}: Assertion .pos < index->n_def. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353 after update
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/rem/rem0rec.cc:1192: ulint rec_get_converted_size_comp_prefix_low(const dict_index_t*, const dfield_t*, ulint, ulint*, rec_comp_status_t, bool) [with bool mblob = false; ulint = long unsigned int]: Assertion `len <= field->col->len || ((field->col->mtype) == 5 || (field->col->mtype) == 14) || (field->col->len == 0 && field->col->mtype == 1)' failed.
    [ 'TBR-211'    , 'mysqld: .{1,150}rem0rec.cc:.{1,30} ulint rec_get_converted_size_comp_prefix_low.{1,500}: Assertion .len <= field->col->len \|\| \(\(field->col->mtype\) == 5 \|\| \(field->col->mtype\) == 14\) \|\| \(field->col->len == 0 \&\& field->col->mtype == 1\). failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353 after update
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/btr/btr0bulk.cc:113: dberr_t PageBulk::init(): Assertion `page_dir_get_n_heap(new_page) == 2U' failed.
    [ 'TBR-212'    , 'mysqld: .{1,150}btr0bulk.cc:.{1,30} dberr_t PageBulk::init\(\): Assertion .page_dir_get_n_heap\(new_page\) == 2U. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353 after second update
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/page/page0cur.cc:1182: rec_t* page_cur_insert_rec_low(const page_cur_t*, dict_index_t*, const rec_t*, offset_t*, mtr_t*): Assertion `current_rec != insert_buf' failed.
    [ 'TBR-213'    , 'mysqld: .{1,150}page0cur.cc:.{1,30} rec_t\* page_cur_insert_rec_low.{1,500}: Assertion .current_rec \!= insert_buf. failed.' ],
    # 2020-02-07 bb-10.5-MDEV-12353 after second update   outside of InnoDB
    # mysqld: /work/bb-10.5-MDEV-12353/sql/sql_error.cc:335: void Diagnostics_area::set_ok_status(ulonglong, ulonglong, const char*): Assertion `!is_set() || (m_status == DA_OK_BULK && is_bulk_op())' failed.
    [ 'TBR-214'    , 'mysqld: .{1,150}sql_error.cc:.{1,30} void Diagnostics_area::set_ok_status.{1,500}: Assertion .\!is_set\(\) \|\| \(m_status == DA_OK_BULK \&\& is_bulk_op\(\)\). failed.' ],
    # seen in 10.5
    [ 'TBR-215'    , 'InnoDB: Index .{1,32} is marked as corrupted' ],
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/handler/handler0alter.cc:1448: bool instant_alter_column_possible(const dict_table_t&, const Alter_inplace_info*, const TABLE*, const TABLE*, bool): Assertion `!pk->has_virtual()' failed.
    [ 'TBR-216'    , 'mysqld: .{1,150}handler0alter.cc:.{1,20} bool instant_alter_column_possible.{1,300}: Assertion .\!pk->has_virtual\(\). failed.' ],
    # outside of innodb
    # # 2020-02-10T19:46:13 [251883] #3  <signal handler called>
    # 2020-02-10T19:46:13 [251883] #4  0x0000564aa3ca4135 in base_list_iterator::next_fast (this=0x7f7b4c15e1a0) at /work/bb-10.5-MDEV-12353/sql/sql_list.h:443
    # 2020-02-10T19:46:13 [251883] #5  0x0000564aa3dc8855 in List_iterator_fast<Item_equal>::operator++ (this=0x7f7b4c15e1a0) at /work/bb-10.5-MDEV-12353/sql/sql_list.h:620
    # 2020-02-10T19:46:13 [251883] #6  0x0000564aa3e5099e in substitute_for_best_equal_field
    [ 'TBR-217'    , '<signal handler called>.{1,500}base_list_iterator::next_fast.{1,500}List_iterator_fast.{1,500}substitute_for_best_equal_field' ],
    # 2020-02-11
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/page/page0cur.cc:1110: rec_t* page_cur_insert_rec_low(const page_cur_t*, dict_index_t*, const rec_t*, offset_t*, mtr_t*): Assertion `cur->rec != insert_buf' failed.
    [ 'TBR-218'    , 'mysqld: .{1,150}page0cur.cc:.{1,20} rec_t\* page_cur_insert_rec_low.{1,300}: Assertion .cur->rec \!= insert_buf. failed.' ],
    # # 2020-02-11T10:37:45 [310235] #3  <signal handler called>
    # 2020-02-11T10:37:45 [310235] #4  0x00005630ca7cc3c9 in buf_page_get_state (bpage=0x7fb654009f20) at /work/bb-10.5-MDEV-12353/storage/innobase/include/buf0buf.ic:220
    # 2020-02-11T10:37:45 [310235] #5  0x00005630ca7cc40e in buf_block_get_state (block=0x7fb654009f20) at /work/bb-10.5-MDEV-12353/storage/innobase/include/buf0buf.ic:249
    # 2020-02-11T10:37:45 [310235] #6  0x00005630ca7ecfd2 in btr_blob_free
    [ 'TBR-220'    , '<signal handler called>.{1,500}buf_page_get_state.{1,500}buf_block_get_state.{1,500}btr_blob_free' ],
    # mysqld: /home/matthias/Server/bb-10.5-MDEV-12353/storage/innobase/row/row0log.cc:2450: const mrec_t* row_log_table_apply_op(que_thr_t*, ulint, row_merge_dup_t*, dberr_t*, mem_heap_t*, mem_heap_t*, const mrec_t*, const mrec_t*, offset_t*): Assertion `0' failed.
    [ 'TBR-222' , 'mysqld: .{1,150}row0log.cc:.{1,20} const mrec_t\* row_log_table_apply_op.{1,300}: Assertion .0. failed.' ],
    # mysqld: /home/matthias/Server/bb-10.5-MDEV-12353/storage/innobase/dict/dict0dict.cc:4536: void dict_table_check_for_dup_indexes(const dict_table_t*, check_name): Assertion `index1->is_committed() != index2->is_committed() || strcmp(index1->name, index2->name) != 0' failed.
    [ 'TBR-223' , 'mysqld: .{1,150}dict0dict.cc:.{1,20} void dict_table_check_for_dup_indexes.{1,300}: Assertion .index1->is_committed\(\) \!= index2->is_committed\(\) \|\| strcmp\(index1->name, index2->name\) != 0. failed.' ],
    # InnoDB: Failing assertion: recv_recovery_is_on() || page_zip_validate( &bpage->zip, page, ((buf_block_t*) bpage)->index)
    [ 'TBR-224'     , 'InnoDB: Failing assertion: recv_recovery_is_on\(\) \|\| page_zip_validate\( \&bpage->zip, page, \(\(buf_block_t\*\) bpage\)->index\)' ],
    # InnoDB: Failing assertion: (block)->index || (block)->n_pointers == 0
    [ 'TBR-226' , 'InnoDB: Failing assertion: \(block\)->index \|\| \(block\)->n_pointers == 0.' ],
    # Outside of InnoDB
    # 2020-02-14T17:34:57 [21095] #3  <signal handler called>
    # 2020-02-14T17:34:57 [21095] #4  __memmove_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:303
    # 2020-02-14T17:34:57 [21095] #5  0x000055a8b042625a in my_copy_8bit (cs=0x55a8b10737a0 <my_charset_latin1>, dst=0x7f26141f9158 "", dst_length=8, src=0x8f00007f26142520 <error: Cannot access memory at address 0x8f00007f26142520>, src_length=8, nchars=8, status=0x7f2698151a30) at /home/mleich/Server/10.4/strings/ctype-simple.c:1169
    # 2020-02-14T17:34:57 [21095] #6  0x000055a8af81534f in String_copier::well_formed_copy (this=0x7f2698151a30, to_cs=0x55a8b10737a0 <my_charset_latin1>, to=0x7f26141f9158 "", to_length=8, from_cs=0x55a8b10737a0 <my_charset_latin1>, from=0x8f00007f26142520 <error: Cannot access memory at address 0x8f00007f26142520>, from_length=8, nchars=8) at /home/mleich/Server/10.4/sql/sql_string.cc:1079
    # 2020-02-14T17:34:57 [21095] #7  0x000055a8afa8a99f in Field_longstr::well_formed_copy_with_check (this=0x7f260c07fae0, to=0x7f26141f9158 "", to_length=8, from_cs=0x55a8b10737a0 <my_charset_latin1>, from=0x8f00007f26142520 <error: Cannot access memory at address 0x8f00007f26142520>, from_length=8, nchars=8, count_spaces=true, copy_length=0x7f2698151aa0) at /home/mleich/Server/10.4/sql/field.h:1916
    # 2020-02-14T17:34:57 [21095] #8  0x000055a8afa7d623 in Field_blob::store
    [ 'TBR-227'  , '<signal handler called>.{1,150}__memmove_avx_unaligned_erms.{1,800}my_copy_8bit.{1,800}String_copier::well_formed_copy' ],
    # SEGV
    # 2020-02-14T18:24:17 [30655] #3  <signal handler called>
    # 2020-02-14T18:24:17 [30655] #4  0x00005649959108b6 in dict_index_t::is_ibuf (this=0x0) at /home/mleich/Server/10.4/storage/innobase/include/dict0mem.h:1165
    # 2020-02-14T18:24:17 [30655] #5  0x0000564995b66a87 in btr_search_hash_table_validate (hash_table_id=2) at /home/mleich/Server/10.4/storage/innobase/btr/btr0sea.cc:2047
    # 2020-02-14T18:24:17 [30655] #6  0x0000564995b670da in btr_search_validate
    [ 'TBR-228'    , '<signal handler called>.{1,150}dict_index_t::is_ibuf.{1,800}btr_search_hash_table_validate.{1,800}btr_search_validate' ],
    # InnoDB: Failing assertion: addr.page == FIL_NULL || addr.boffset >= FIL_PAGE_DATA
    [ 'TBR-229'    , 'InnoDB: Failing assertion: addr.page == FIL_NULL \|\| addr.boffset >= FIL_PAGE_DATA' ],
    # 2020-02-14T19:45:16 [24149] #3  <signal handler called>
    # 2020-02-14T19:45:16 [24149] #4  __memmove_avx_unaligned_erms () at ../sysdeps/x86_64/multiarch/memmove-vec-unaligned-erms.S:303
    # 2020-02-14T19:45:16 [24149] #5  0x000055c681999dbd in mem_heap_dup (heap=0x7f2dcc066170, data=0x8f8f8f8f8f8f8f8f, len=9) at /home/mleich/Server/10.4/storage/innobase/include/mem0mem.h:242
    # 2020-02-14T19:45:16 [24149] #6  0x000055c681afebcb in dfield_dup (field=0x55c6858d3a50, heap=0x7f2dcc066170) at /home/mleich/Server/10.4/storage/innobase/include/data0data.ic:182
    # 2020-02-14T19:45:16 [24149] #7  0x000055c681b061fc in row_merge_buf_add
    [ 'TBR-230'    , '<signal handler called>.{1,150}__memmove_avx_unaligned_erms.{1,800}mem_heap_dup.{1,800}dfield_dup' ],
    # actual 10.5 but not InnoDB
    # 2020-02-18T13:42:41 [180191] #3  <signal handler called>
    # 2020-02-18T13:42:41 [180191] #4  0x00007f47f3d9f2a0 in __memmove_ssse3_back () from /lib64/libc.so.6
    # 2020-02-18T13:42:41 [180191] #5  0x000055711349893a in my_copy_8bit (cs=0x557114681da0 <my_charset_latin1>, dst=0x7f45440946b8 "", dst_length=499, src=0x6100007f45440601 <Address 0x6100007f45440601 out of bounds>, src_length=499, nchars=499, status=0x7f47f04b3dd0) at /home/matthias/Server/10.5/strings/ctype-simple.c:1169
    # 2020-02-18T13:42:41 [180191] #6  0x0000557112d63833 in charset_info_st::copy_fix (this=0x557114681da0 <my_charset_latin1>, dst=0x7f45440946b8 "", dst_length=499, src=0x6100007f45440601 <Address 0x6100007f45440601 out of bounds>, src_length=499, nchars=499, status=0x7f47f04b3dd0) at /home/matthias/Server/10.5/include/m_ctype.h:751
    # 2020-02-18T13:42:41 [180191] #7  0x0000557112d63152 in String_copier::well_formed_copy
    [ 'TBR-231'    , '<signal handler called>.{1,500}__memmove_ssse3_back.{1,500}my_copy_8bit.{1,500}charset_info_st::copy_fix' ],

    [ 'MDEV-16686' , 'mysqld: .{1,150}lock.cc:.{1,30} int lock_external\(THD\*, TABLE\*\*, uint\): Assertion .\(\*tables\)->reginfo.lock_type >= TL_READ. failed.' ],
    # closed
    # [ 'MDEV-17843' , 'mysqld: .{1,150}lock0lock.cc:.{1,30} bool lock_rec_queue_validate\(.{1,250}\): Assertion .page_rec_is_leaf\(rec\). failed.' ],
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/row/row0log.cc:3076: dberr_t row_log_table_apply_ops(que_thr_t*, row_merge_dup_t*, ut_stage_alter_t*): Assertion `0' failed.
    [ 'MDEV-18361' , 'mysqld: .{1,150}row0log.cc:.{1,20} dberr_t row_log_table_apply_ops.{1,300}: Assertion .0. failed.' ],
    [ 'MDEV-19555' , 'mysqld: .{1,150}sql_error.h:.{1,30} uint Diagnostics_area::sql_errno\(\) const: Assertion .m_status == DA_ERROR. failed.' ],
    [ 'MDEV-20038' , 'mysqld: .{1,150}btr0cur.cc:.{1,20} dberr_t btr_cur_search_to_nth_level_func.{1,300}: Assertion `rw_lock_own\(dict_index_get_lock\(index\), RW_LOCK_S\)\' failed.' ],
    [ 'MDEV-20618' , 'mysqld: .{1,150}row0upd.cc:.{1,30} dberr_t row_upd_sec_index_entry.{1,100}\): Assertion .btr_validate_index\(index, 0\). failed.' ],
    [ 'MDEV-21564-Rebirth' , 'mysqld: .{1,150}trx0purge.cc:.{1,30} void trx_purge_add_undo_to_history\(.{1,100}\): Assertion .srv_undo_sources \|\| trx->undo_no == 0 \|\| \(\!purge_sys.enabled\(\) .. \(srv_is_being_started \|\| trx_rollback_is_active \|\| srv_force_recovery >= SRV_FORCE_NO_BACKGROUND\)\) \|\| \(\(trx->mysql_thd \|\| trx->internal\) .. srv_fast_shutdown\). failed.' ],
    # mysqld: /work/10.5/storage/innobase/dict/dict0load.cc:1956: void dict_load_virtual_one_col(dict_table_t*, ulint, dict_v_col_t*, mem_heap_t*): Assertion `pos == vcol_pos' failed.
    [ 'MDEV-21663' , 'mysqld: .{1,150}dict0load.cc:.{1,30} void dict_load_virtual_one_col.{1,300}: Assertion .pos == vcol_pos. failed.' ],
    # mysqld: /home/matthias/Server/bb-10.5-MDEV-12353/storage/innobase/page/page0cur.cc:1306: rec_t* page_cur_insert_rec_low(const page_cur_t*, dict_index_t*, const rec_t*, offset_t*, mtr_t*): Assertion `rdm - rd + bd <= insert_buf + rec_size' failed.
    [ 'MDEV-21749' , 'mysqld: .{1,150}page0cur.cc:.{1,20} rec_t\* page_cur_insert_rec_low.{1,300}: Assertion .rdm - rd \+ bd <= insert_buf \+ rec_size.' ],
    # mysqld: /work/bb-10.5-MDEV-12353/storage/innobase/page/page0cur.cc:1312: rec_t* page_cur_insert_rec_low(const page_cur_t*, dict_index_t*, const rec_t*, offset_t*, mtr_t*): Assertion `!memcmp(insert_buf, rec - extra_size, rec_size)' failed.
    [ 'MDEV-21779' , 'mysqld: .{1,150}page0cur.cc:.{1,20} rec_t\* page_cur_insert_rec_low.{1,300}: Assertion .\!memcmp\(insert_buf, rec - extra_size, rec_size\). failed.' ],
    # mysqld: /work/10.5/storage/innobase/dict/dict0load.cc:1650: const char* dict_load_column_low(dict_table_t*, mem_heap_t*, dict_col_t*, table_id_t*, const char**, const rec_t*, ulint*): Assertion `vcol->v_pos == dict_get_v_col_pos(pos)' failed.
    [ 'MDEV-21800' , 'mysqld: .{1,150}dict0load.cc:.{1,30} const char\* dict_load_column_low.{1,300}: Assertion .vcol->v_pos == dict_get_v_col_pos\(pos\). failed.' ],

],

'search_var_size' => 30000000,

}



